jax/_src/pallas/core.py

```diff
@@ -65,6 +65,10 @@ def __repr__(self):
 SEMAPHORE_INTERPRET_DTYPE = jnp.int16
 SEMAPHORE_MAX_VALUE = jnp.iinfo(SEMAPHORE_INTERPRET_DTYPE).max
 
+class semaphore_dtype(dtypes.extended): pass
+class semaphore(semaphore_dtype): pass
+class dma_semaphore(semaphore_dtype): pass
+class barrier_semaphore(semaphore_dtype): pass
```

Comment:
@apaszke I kept all the types since they are used in `check_sem_avals` in the same file. Let me if somehow you want to move part of this check in the mosaic lowering.

Reply:
I think the right thing to do here would be to keep `dma_semaphore` under `pltpu`, and just make the list of allowed semaphore dtypes explicit in every call to `check_sem_avals` (which can live in this file but continue to be used in Pallas:TPU backend).